### ISSUE TITLE
Enhancement: Have an <extraFiles> option to add individual files instead of a whole directory

### ISSUE THREAD

**Issue by MiellyEllen** [GoogleContainerTools/jib#3164]
**Environment**:

- *Jib version:*2.8.0
- *Build tool:* mvn
- *OS:* Windows

**Description of the issue**:
If this does not exist already, could there be an enhancement to add individual files into an image layer, rather than selecting all the files in a directory.

**Expected behavior**:
```
<extraFiles>
   <extraFile><from>../folder1/context1.xml</from><to>/app/contexts</to></extrafile>
<extraFiles>
```

**Additional Information**: If this is not available, is there an expected version where this might be added?

**Comment by chanseokoh** [comment]
Thanks for the feedback! Yeah, unfortunately `<extraDirectories>` requires preparing a directory. If you can just put the file at `src/main/jib/app/contexts/context1.xml`, the file will be copied into the image (as `/src/main/jib` is the default extra directory).

If you can't put it in `src/main/jib` (e.g., you don't want to duplicate the file, or you only want to filter out a single file in a directory), it's still easy to prepare a directory in Maven. Here's an example using `maven-resources-plugin`, which prepares the directory `target/jib-extras` to be used for `<extraDirectories>`:

```xml
      <plugin>
        <artifactId>maven-resources-plugin</artifactId>
        <version>3.2.0</version>
        <executions>
          <execution>
            <id>copy-extras</id>
            <phase>process-resources</phase>
            <goals>
              <goal>copy-resources</goal>
            </goals>
            <configuration>
              <outputDirectory>target/jib-extras</outputDirectory>
              <resources>
                <resource>
                  <directory>target/some-files</directory>
                  <includes>**/*.txt,**/*.rtf</includes>
                </resource>
              </resources>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

Then for the Jib configuration,
```xml
          <extraDirectories>
            <paths>
              <path>
                <from>target/jib-extras</from>
                <into>/dest-dir</into>
              </path>
            </paths>
          </extraDirectories>
```

I'm not saying that adding the `<extraFiles>` feature doesn't provide a value; I think it's useful in many situations. It's just that it will be a low priority for us, and there's a workaround.

But probably, instead of adding the new config `<extraFiles>`, what we could do is to add a filtering mechanism similar to `maven-resources-plugin`. For example,
```xml
          <extraDirectories>
            <paths>
              <path>
                <from>../folder1/</from>
                <into>/app/contexts</into>
                <includes>context1.xml</includes>
              </path>
            </paths>
          </extraDirectories>
```
That way, we can have the same effect of copying a single file without adding `<extraFiles>`. What's even better is that it's more general and flexible and enables copying multiple files.

**Comment by chanseokoh** [comment]
Actually, we have an open issue. Closing this as a dup of #2564. Please follow #2564.

**Comment by MiellyEllen** [comment]
Thanks so much for the very helpful feedback as well as the priority insight. 
Cheers!

**Comment by chanseokoh** [comment]
@tomikmar @labmonkey @andritch @daanschipper @cpvandehey @Nemesys @MiellyEllen

We released Jib plugins 3.0 that implemented a inclusion/exclusion feature to `<extraDirectories>`, which not only makes it super easy to add a single file but is very flexible in copying a subset of files. Check out the docs ([Gradle](https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin/README.md#adding-arbitrary-files-to-the-image) / [Maven](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/README.md#adding-arbitrary-files-to-the-image)) for example usage.
